netstandard/tools/RemoveDepsFileConflicts.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Extensions.DependencyModel;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class RemoveDepsFileConflicts : Task
+    {
+        Dictionary<string, HashSet<string>> compileConflicts = new Dictionary<string, HashSet<string>>(StringComparer.OrdinalIgnoreCase);
+        Dictionary<string, HashSet<string>> runtimeConflicts = new Dictionary<string, HashSet<string>>(StringComparer.OrdinalIgnoreCase);
+
+        [Required]
+        public string DepsFilePath { get; set; }
+
+        [Required]
+        public ITaskItem[] Conflicts { get; set; }
+
+        public override bool Execute()
+        {
+            LoadConflicts();
+
+            DependencyContext sourceDeps;
+            using (var sourceStream = File.Open(DepsFilePath, FileMode.Open, FileAccess.Read, FileShare.Read | FileShare.Delete))
```

Comment:
Pardon my ignorance on this topic, but it seems you are just opening a file to read its contents and then closing it immediately. Why wouldn't the convenience function "do the right thing" in this case? Is your scenario special in some way that I'm missing?  Or does the convenience function not "do the right thing" for the 80% case (which to me, it appears your scenario falls into the 80% case).